### PR TITLE
Cleaned Up Word Commands

### PR DESCRIPTION
**Pull request by openbots-ff**

### CODE REVIEW

File: taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs [hunk 1]
```diff
@@ -1,137 +1,145 @@
...
-        [Attributes.PropertyAttributes.InputSpecification("Specify the font name.")]
-        [Attributes.PropertyAttributes.SampleUsage("Select **Arial**")]
-        [Attributes.PropertyAttributes.Remarks("")]
+        [PropertyDescription("Font Name")]
+        [PropertyUISelectionOption("Arial")]
+        [PropertyUISelectionOption("Calibri")]
+        [PropertyUISelectionOption("Helvetica")]
+        [PropertyUISelectionOption("Times New Roman")]
+        [PropertyUISelectionOption("Verdana")]
+        [InputSpecification("Select or provide a valid font name.")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs:47]: I think It should simply be like "Select a font name."

File: taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs [hunk 2]
```diff
@@ -1,137 +1,145 @@
...
-        [Attributes.PropertyAttributes.Remarks("")]
+        [PropertyDescription("Font Size")]
+        [PropertyUISelectionOption("10")]
+        [PropertyUISelectionOption("11")]
+        [PropertyUISelectionOption("12")]
+        [PropertyUISelectionOption("14")]
+        [PropertyUISelectionOption("16")]
+        [PropertyUISelectionOption("18")]
+        [PropertyUISelectionOption("20")]
+        [InputSpecification("Select or provide a valid font size.")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs:61]: Similarly, It should be "Select a font size."

File: taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs [hunk 3]
```diff
@@ -142,9 +150,10 @@ public override List<Control> Render(frmCommandEditor editor)
 
             return RenderedControls;
         }
+
         public override string GetDisplayValue()
         {
-            return base.GetDisplayValue() + " ['" + v_TextToSet + "' To Instance Name: '" + v_InstanceName + "']";
+            return base.GetDisplayValue() + $" [Append '{v_TextToSet}' - Instance Name: '{v_InstanceName}']";
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs]: Remove colon from **Instance Name:**

File: taskt/Core/Automation/Commands/Word Commands/WordAppendImageCommand.cs [hunk 4]
```diff
@@ -62,19 +65,20 @@ public override void RunCommand(object sender)
...
             RenderedControls.AddRange(CommandControls.CreateDefaultInputGroupFor("v_InstanceName", this, editor));
             RenderedControls.AddRange(CommandControls.CreateDefaultInputGroupFor("v_ImagePath", this, editor));
 
             return RenderedControls;
         }
+
         public override string GetDisplayValue()
         {
-            return base.GetDisplayValue() + " ['" + v_ImagePath + "' To Instance Name: '" + v_InstanceName + "']";
+            return base.GetDisplayValue() + $" [Append '{v_ImagePath}' - Instance Name: '{v_InstanceName}']";
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Word Commands/WordAppendImageCommand.cs]: Remove colon

File: taskt/Core/Automation/Commands/Word Commands/WordAppendDataTableCommand.cs [hunk 5]
```diff
@@ -108,45 +110,44 @@ public override void RunCommand(object sender)
...
+
+            RenderedControls.AddRange(CommandControls.CreateDefaultInputGroupFor("v_InstanceName", this, editor));
+            RenderedControls.AddRange(CommandControls.CreateDefaultInputGroupFor("v_DataTable", this, editor));
 
+            return RenderedControls;
+        }
+
+        public override string GetDisplayValue()
+        {
+            return base.GetDisplayValue() + $" [Append '{v_DataTable}' - Instance Name: '{v_InstanceName}']";
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Word Commands/WordAppendDataTableCommand.cs]: Remove colon

File: taskt/Core/Automation/Commands/Word Commands/WordCreateApplicationCommand.cs [hunk 6]
```diff
@@ -2,64 +2,75 @@
...
 
     [Serializable]
-    [Attributes.ClassAttributes.Group("Word Commands")]
-    [Attributes.ClassAttributes.Description("This command creates a Word Application.")]
-    [Attributes.ClassAttributes.UsesDescription("Use this command when you want to launch a new instance of Word.")]
-    [Attributes.ClassAttributes.ImplementationDescription("This command implements Word Interop to achieve automation.")]
+    [Group("Word Commands")]
+    [Description("This command creates a Word Instance.")]
+    [UsesDescription("Use this command when you want to launch a new instance of Word.")]
+    [ImplementationDescription("This command implements Word Interop to achieve automation.")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Word Commands/WordCreateApplicationCommand.cs]: Remove **UsesDescription** and **ImplementationDescription** attributes

File: taskt/Core/Automation/Commands/Word Commands/WordCreateApplicationCommand.cs [hunk 7]
```diff
@@ -2,64 +2,75 @@
...
+        [SampleUsage("MyWordInstance")]
+        [Remarks("This unique name allows you to refer to the instance by name in future commands, " +
+                 "ensuring that the commands you specify run against the correct application.")]
         public string v_InstanceName { get; set; }
 
+        [XmlAttribute]
+        [PropertyDescription("Visible")]
+        [PropertyUISelectionOption("Yes")]
+        [PropertyUISelectionOption("No")]
+        [InputSpecification("Indicate whether Excel automation should be visible or not.")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Word Commands/WordCreateApplicationCommand.cs]: Please make the suggested change.
```suggestion
        [InputSpecification("Indicate whether the Word automation should be visible or not.")]
```

File: taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs (hunk 1 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs:47]: The thing is that these aren't the only options. You can use any font provided by Word. I just added the most common ones.

File: taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs (hunk 2 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs:61]: Same here. Font number can be anything.

File: taskt/Core/Automation/Commands/Word Commands/WordReplaceTextCommand.cs [hunk 8]
```diff
@@ -61,7 +61,7 @@ public override void RunCommand(object sender)
             var wordObject = engine.GetAppInstance(vInstance);
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Word Commands/WordReplaceTextCommand.cs:62]: Please update a comment on line-60 to "//get word app object"

File: taskt/Core/Automation/Commands/Word Commands/WordAddDocumentCommand.cs [hunk 9]
```diff
@@ -2,58 +2,60 @@
...
+            CommandEnabled = true;
+            CustomRendering = true;
         }
+
         public override void RunCommand(object sender)
         {
-            var engine = (Core.Automation.Engine.AutomationEngineInstance)sender;
+            var engine = (AutomationEngineInstance)sender;
             var vInstance = v_InstanceName.ConvertToUserVariable(engine);
             var excelObject = engine.GetAppInstance(vInstance);
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Word Commands/WordAddDocumentCommand.cs]: please change `excelObject` to `wordObject`.
Make sure to update its references in the code.

File: taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs (hunk 1 above)
Comment (ob-ab) [reply taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs:47]: got it

File: taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs (hunk 2 above)
Comment (ob-ab) [reply taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs:61]: got it

**Review by ob-ab — CHANGES_REQUESTED** [review]
Please make a few small changes.

File: taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs (hunk 3 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Word Commands/WordAppendTextCommand.cs]: done

File: taskt/Core/Automation/Commands/Word Commands/WordAppendImageCommand.cs (hunk 4 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Word Commands/WordAppendImageCommand.cs]: done

File: taskt/Core/Automation/Commands/Word Commands/WordAppendDataTableCommand.cs (hunk 5 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Word Commands/WordAppendDataTableCommand.cs]: done

File: taskt/Core/Automation/Commands/Word Commands/WordCreateApplicationCommand.cs (hunk 6 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Word Commands/WordCreateApplicationCommand.cs]: done

File: taskt/Core/Automation/Commands/Word Commands/WordReplaceTextCommand.cs (hunk 8 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Word Commands/WordReplaceTextCommand.cs:62]: done

File: taskt/Core/Automation/Commands/Word Commands/WordAddDocumentCommand.cs (hunk 9 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Word Commands/WordAddDocumentCommand.cs]: done

**Review by ob-ab — APPROVED** [review]
Looks good to me.

**Review by CameronHerwig — APPROVED** [review]

**Pull request merged**